lib/pbench/server/api/__init__.py

```diff
@@ -20,91 +23,138 @@
+        if not prdp:
+            app.logger.error("Missing config variable for pbench-receive-dir-prefix")
+            sys.exit(1)
+        try:
+            upload_directory = Path(f"{prdp}-002").resolve(strict=True)
+        except FileNotFoundError:
+            app.logger.exception("pbench-receive-dir-prefix does not exist on the host")
+            sys.exit(1)
+        except Exception:
+            app.logger.exception(
+                "Exception occurred during setting up the upload directory on the host"
+            )
+            sys.exit(1)
+        else:
+            app.upload_directory = upload_directory
 
-def create_app():
-    """Create Flask app with defined resource endpoints."""
+        app.config["PORT"] = app.config_server.get("rest_port")
+        app.config["VERSION"] = app.config_server.get("rest_version")
+        app.config["MAX_CONTENT_LENGTH"] = filesize_bytes(
+            app.config_server.get("rest_max_content_length")
+        )
+        app.config["REST_URI"] = app.config_server.get("rest_uri")
+        app.config["LOG"] = app.config_server.get("rest_log")
+        app.config["BIND_HOST"] = app.config_server.get("bind_host")
+        app.config["WORKERS"] = app.config_server.get("workers")
 
-    global app
+        # postgresql specific configuration
+        app.config["DB_SECRET_KEY"] = os.getenv("SECRET_KEY", "my_precious")
+        app.config["BCRYPT_LOG_ROUNDS"] = app.config_server.get("bycrypt_log_rounds")
+        app.config["POSTGRES_DB_URI"] = app.config_server.get("postgres_db_uri")
 
-    cfg_name = os.environ.get("_PBENCH_SERVER_CONFIG")
-    if not cfg_name:
-        print(
-            f"{__name__}: ERROR: No config file specified; set"
-            " _PBENCH_SERVER_CONFIG",
-            file=sys.stderr,
-        )
-        sys.exit(1)
-
-    try:
-        config = PbenchServerConfig(cfg_name)
-    except BadConfig as e:
-        print(f"{__name__}: {e} (config file {cfg_name})", file=sys.stderr)
-        sys.exit(1)
-
-    app = Flask(__name__)
-    api = Api(app)
-
-    app.logger = get_pbench_logger(__name__, config)
-    app.config_server = config.conf["pbench-server"]
-    app.config_elasticsearch = config.conf["elasticsearch"]
-    app.config_graphql = config.conf["graphql"]
-
-    prdp = app.config_server.get("pbench-receive-dir-prefix")
-    if not prdp:
-        app.logger.error("Missing config variable for pbench-receive-dir-prefix")
-        sys.exit(1)
-    try:
-        upload_directory = Path(f"{prdp}-002").resolve(strict=True)
-    except FileNotFoundError:
-        app.logger.exception("pbench-receive-dir-prefix does not exist on the host")
-        sys.exit(1)
-    except Exception:
-        app.logger.exception(
-            "Exception occurred during setting up the upload directory on the host"
-        )
-        sys.exit(1)
-    else:
-        app.upload_directory = upload_directory
+        bcrypt = Bcrypt(app)
+        db = SQLAlchemy(app)
+        migrate = Migrate(db)
```

Comment:
I still don't like exposing all this user management mechanism outside the user classes. It would be cleaner to have it all encapsulated in a class. (Even if it requires "cheating" by setting more properties on the app object.)